uuid.go

```diff
@@ -4,22 +4,36 @@ import (
 	"crypto/rand"
 	"encoding/hex"
 	"fmt"
+	"io"
 )
 
 // GenerateRandomBytes is used to generate random bytes of given size.
 func GenerateRandomBytes(size int) ([]byte, error) {
+	return generateRandomBytesByReader(size, rand.Reader)
+}
+
+func generateRandomBytesByReader(size int, reader io.Reader) ([]byte, error) {
 	buf := make([]byte, size)
-	if _, err := rand.Read(buf); err != nil {
+	if _, err := io.ReadFull(reader, buf); err != nil {
 		return nil, fmt.Errorf("failed to read random bytes: %v", err)
 	}
 	return buf, nil
 }
 
+
 const uuidLen = 16
 
 // GenerateUUID is used to generate a random UUID
 func GenerateUUID() (string, error) {
-	buf, err := GenerateRandomBytes(uuidLen)
+	return GenerateUUIDByReader(rand.Reader)
+}
+
+// GenerateUUIDByReader is used to generate a random UUID by a given Reader
+func GenerateUUIDByReader(reader io.Reader) (string, error) {
```

Comment:
nit: consider FromReader or WithReader. I think that's a little more clear from a naming perspective